staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -39,6 +40,583 @@ func RunTestWatch(ctx context.Context, t *testing.T, store storage.Interface) {
+				},
+				IndexFields: []string{"spec.nodeName"},
+			},
+		},
+		{
+			name:          "cluster-wide watch, request with name, field selecotr with metadata.name",
+			requestedName: "foo5",
+			watchTests: []*testWatchStruct{
+				{basePod("foo5"), true, watch.Added},
+				{basePod("foo6"), false, ""},
+				{basePodUpdated("foo5"), true, watch.Modified},
+				{basePodUpdated("foo6"), false, ""},
+			},
+			pred: storage.SelectionPredicate{
+				Label:    labels.Everything(),
+				Field:    fields.ParseSelectorOrDie("metadata.name=foo5"),
+				GetAttrs: storage.DefaultClusterScopedAttr,
+			},
+		},
+		{
+			name:          "cluster-wide watch, request with name, field selecotr with metadata.name and spec.nodeName",
+			requestedName: "foo6",
+			watchTests: []*testWatchStruct{
+				{basePod("foo6"), false, ""},
+				{basePod("foo7"), false, ""},
+				{basePodUpdated("foo6"), false, ""},
+				{basePodUpdated("foo6"), false, ""},
+				{basePodAssigned("foo6", "bar4"), true, watch.Added},
+			},
+			pred: storage.SelectionPredicate{
+				Label: labels.Everything(),
+				Field: fields.SelectorFromSet(fields.Set{
+					"metadata.name": "foo6",
+					"spec.nodeName": "bar4",
+				}),
+				GetAttrs: func(obj runtime.Object) (labels.Set, fields.Set, error) {
+					pod := obj.(*example.Pod)
+					return nil, fields.Set{"spec.nodeName": pod.Spec.NodeName, "metadata.name": pod.ObjectMeta.Name}, nil
+				},
+				IndexFields: []string{"spec.nodeName"},
+			},
+		},
+		{
+			name:          "cluster-wide watch, request with name, field selecotr with metadata.name, and with spec.nodeName to filter out watch",
+			requestedName: "foo8",
+			watchTests: []*testWatchStruct{
+				{basePod("foo8"), true, watch.Added},
+				{basePod("foo9"), false, ""},
+				{basePodUpdated("foo8"), true, watch.Modified},
+				{basePodAssigned("foo8", "bar5"), true, watch.Deleted},
+				{basePodAssigned("foo9", "bar5"), false, ""},
+			},
+			pred: storage.SelectionPredicate{
+				Label: labels.Everything(),
+				Field: fields.AndSelectors(fields.ParseSelectorOrDie("spec.nodeName!=bar5"), fields.SelectorFromSet(fields.Set{"metadata.name": "foo8"})),
+				GetAttrs: func(obj runtime.Object) (labels.Set, fields.Set, error) {
+					pod := obj.(*example.Pod)
+					return nil, fields.Set{"spec.nodeName": pod.Spec.NodeName, "metadata.name": pod.ObjectMeta.Name}, nil
+				},
+				IndexFields: []string{"spec.nodeName"},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			requestInfo := &genericapirequest.RequestInfo{}
+			requestInfo.Name = tt.requestedName
+			requestInfo.Namespace = ""
+			ctx = genericapirequest.WithRequestInfo(ctx, requestInfo)
+			ctx = genericapirequest.WithNamespace(ctx, "")
+
+			watchKey := "/pods"
+			if tt.requestedName != "" {
+				watchKey += "/" + tt.requestedName
+			}
+
+			w, err := store.Watch(ctx, watchKey, storage.ListOptions{ResourceVersion: "0", Predicate: tt.pred, Recursive: tt.recursive})
```

Comment:
I'm not a fan of using ResourceVersion: "0" here.

Can you instead just do:
- list of all pods here
- use the RV that you get from list

[Similarly as it is done in the existing testWatch test].

Reply:
Revised as suggested. Thank you!